cmd/testsign.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright © 2022 IAB Technology Laboratory, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"github.com/spf13/cobra"
+	"time"
+
+	"github.com/IABTechLab/adscert/pkg/adscert/api"
+	"github.com/IABTechLab/adscert/pkg/adscert/logger"
+	"github.com/IABTechLab/adscert/pkg/adscert/signatory"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/protobuf/encoding/prototext"
+)
+
+// testsignCmd represents the test sign command
+var (
+	testsignParams = &testsignParameters{url: "google.com"}
+
+	testsignCmd = &cobra.Command{
+		Use:   "testsign",
+		Short: " Given a URL to invoke, generate a signature.",
+		Run: func(cmd *cobra.Command, args []string) {
+			signRequest(testsignParams)
+		},
+	}
+)
+
+type testsignParameters struct {
+	url            string
+	serverAddress  string
+	destinationURL string
+	body           string
+	signingTimeout time.Duration
+}
+
+func init() {
+	rootCmd.AddCommand(testsignCmd)
+
+	testsignCmd.Flags().StringVar(&testsignParams.destinationURL, "url", "https://google.com/", "URL to invoke")
+	testsignCmd.Flags().StringVar(&testsignParams.serverAddress, "server_address", "localhost:3000", "address of grpc server")
+	testsignCmd.Flags().StringVar(&testsignParams.body, "body", "", "POST request body")
+	testsignCmd.Flags().DurationVar(&testsignParams.signingTimeout, "signing_timeout", 5*time.Millisecond, "Specifies how long this client will wait for signing to finish before abandoning.")
+}
+
+func signRequest(testsignParams *testsignParameters) {
+
+	// Establish the gRPC connection that the client will use to connect to the
+	// signatory server.  This basic example uses unauthenticated connections
+	// which should not be used in a production environment.
+	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+	conn, err := grpc.Dial(testsignParams.serverAddress, opts...)
+	if err != nil {
+		logger.Fatalf("Failed to dial: %v", err)
+	}
+	defer conn.Close()
+
+	// Optional: performs a health check against the server before actually
+	// trying to invoke the signatory service.
+
+	performOptionalHealthCheckRPC(conn)
```

Comment:
I think it would be OK to remove this health check code from this command.

Reply:
done!